src/hooks/useAsyncFetch.js

```diff
@@ -0,0 +1,18 @@
+import { useVideos } from "../context/video-context";
+import {useEffect} from "react";
+import axios from "axios";
+
+export function useAsyncFetch({url, dispatchType, dispatchPayload}){
+    const {dispatch}  = useVideos();
```

Comment:
Like how you've made use of custom hooks to make things easier !

Reply:
Thanks